userland/src/parameters.cpp

```diff
@@ -0,0 +1,74 @@
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+template <typename ParameterTypeT>
+ParameterTypeT
+wait_for_future(rclcpp::node::Node::SharedPtr & node, std::shared_future<ParameterTypeT> & future)
+{
+  std::future_status status;
+  do {
+    rclcpp::spin_some(node);
+    status = future.wait_for(std::chrono::milliseconds(100));
```

Comment:
Looks like you can pass a 0 duration or use `wait_util` and give it a time in the past:

http://stackoverflow.com/a/10917945/671658

Also, they are going to add a `.is_ready()` non-blocking check in the future and there might already be a `._is_ready` function. See:

http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n3857.pdf (search for `is_ready`)